src/migrations/20220331085057-add-api-link-table.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+exports.up = function (db, cb) {
+    db.runSql(
+        `
+        CREATE TABLE IF NOT EXISTS api_token_project
+        (
+            secret     text NOT NULL,
+            project    text NOT NULL,
+            FOREIGN KEY (secret) REFERENCES api_tokens (secret) ON DELETE CASCADE,
+            FOREIGN KEY (project) REFERENCES projects(id) ON DELETE CASCADE
+        );
+
+        INSERT INTO api_token_project SELECT secret, project FROM api_tokens WHERE project IS NOT NULL;
+
+        ALTER TABLE api_tokens DROP COLUMN "project";
+      `,
+        cb,
+    );
+};
+
+//This is a lossy down migration, tokens with multiple projects are discarded
```

Comment:
That's fair, I'd be a bit surprised if I had multple project tokens and then ended up with separate tokens for each project if I reverted the thing that allowed me to have a token accessing multiple projects.

Reply:
My thought too! I absolutely don't want them getting more permissive, and if you make them less permissive it's just weird. Clearly to destroy for everyone involved